tls/resource_certificate.go

```diff
@@ -185,6 +199,26 @@ func createCertificate(d *schema.ResourceData, template, parent *x509.Certificat
 	}
 	certPem := string(pem.EncodeToMemory(&pem.Block{Type: pemCertType, Bytes: certBytes}))
 
+	private_key, err := parsePrivateKey(d, "private_key_pem", "key_algorithm")
+	// Set PKCS12 data. This is only set if there is a private key present.
+	if err != nil {
```

Comment:
I think this should be inverted.
You want to go into the `if` block if no error is given.

Reply:
fixed